apps/dapp/src/components/Pages/Core/NewUI/UnstakeModal.tsx

```diff
@@ -27,95 +30,112 @@ export const UnstakeOgtModal: React.FC<IProps> = ({ isOpen, onClose }) => {
     setUnstakeAmount(amount);
   }, [balance]);
 
+  useEffect(() => {
+    const onMount = async () => {
+      if (wallet && signer) await updateLockedEntries();
+    };
+    onMount();
+  }, [wallet, signer]);
+
   const bigAmount = getBigNumberFromString(unstakeAmount || '0');
   const buttonIsDisabled =
     unstakeLoading || !unstakeAmount || balance.ogTemple.lte(ZERO) || bigAmount.gt(balance.ogTemple);
 
+  const handleUnlockOGT = async () => {
+    try {
+      lockedEntries.map(async (entry) => {
+        if (Date.now() > entry.lockedUntilTimestamp) await claimOgTemple(0);
+        updateLockedEntries();
+        updateBalance();
+      });
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
   return (
     <>
       <Popover isOpen={isOpen} onClose={onClose} closeOnClickOutside showCloseButton>
-        <UnstakeOgtContainer>
-          <UnstakeTitle>Unstake OGT</UnstakeTitle>
-          <UnstakeSubtitle>You are eligible to unstake:</UnstakeSubtitle>
-          <TempleAmountContainer>
-            <Temple>$OGTEMPLE</Temple>
-            <TempleAmount>{balance.ogTemple ? formatTemple(balance.ogTemple) : '0.00'}</TempleAmount>
-          </TempleAmountContainer>
-          <UnstakeButton disabled={buttonIsDisabled} onClick={() => unstake(unstakeAmount)}>
-            Unstake
-          </UnstakeButton>
-        </UnstakeOgtContainer>
+        <ModalContainer>
+          <Title>Unstake OGTemple</Title>
+          {/* Display Unlock if they have lockedOGT, then display Unstake */}
+          {lockedEntries.length > 0 ? (
+            <>
+              <Subtitle>You can claim locked OGTemple from the Opening Ceremony:</Subtitle>
+              <ClaimButton
+                isSmall
+                onClick={() => handleUnlockOGT()}
+                disabled={Date.now() < lockedEntries[0].lockedUntilTimestamp}
+              >
+                Claim{' '}
+                {formatTemple(lockedEntries.reduce((sum, cur) => sum.add(cur.balanceOGTemple), BigNumber.from('0')))}{' '}
```

Comment:
Nit: I'd move this above to a named variable to improve readability, e.g. `const claimAmount = formatTemple(...)`.

Then 
```
Claim {claimAmount} OGTemple
```